File: python/web_search.py
"""
Web Search — Real-Time Data
============================
RoutePlex automatically detects when your query needs real-time data and
searches the web before sending to the LLM. No extra setup required.

Triggers include: current events, prices, weather, scores, recent dates.

Docs: https://routeplex.com/docs/examples
Usage: python web_search.py
"""

import os
import requests

API_KEY = os.environ.get("ROUTEPLEX_API_KEY")
BASE_URL = "https://api.routeplex.com"

if not API_KEY:
    print("Error: Set ROUTEPLEX_API_KEY environment variable")
    exit(1)


def search_and_answer(question: str) -> str:
    """Ask a question that requires real-time data. Web search is automatic."""
    response = requests.post(
        f"{BASE_URL}/api/v1/chat",
        headers={"Authorization": f"Bearer {API_KEY}"},
        json={
            "messages": [{"role": "user", "content": question}],
            "mode": "routeplex-ai",
            "strategy": "balanced",
        },
    )

    if response.status_code != 200:
        print(f"Error {response.status_code}: {response.text}")
        return ""

    data = response.json()
    print(f"  Model: {data['data']['model_used']} | Cost: ${data['data']['usage']['cost_usd']}")
    return data["data"]["output"]


def main():
    queries = [
        "What is the current price of Bitcoin today?",
        "What are the latest headlines in tech news?",
    ]

    for q in queries:
        print(f"\nQ: {q}")
        answer = search_and_answer(q)
        print(f"A: {answer}")


if __name__ == "__main__":
    main()

File: javascript/web_search.js
/**
 * Web Search — Real-Time Data
 *
 * RoutePlex auto-detects queries that need real-time data and searches
 * the web before sending to the LLM. No extra setup required.
 *
 * Docs: https://routeplex.com/docs/examples
 * Usage: node web_search.js
 */

const API_KEY = process.env.ROUTEPLEX_API_KEY;
const BASE_URL = "https://api.routeplex.com";

if (!API_KEY) {
  console.error("Error: Set ROUTEPLEX_API_KEY environment variable");
  process.exit(1);
}

async function searchAndAnswer(question) {
  const response = await fetch(`${BASE_URL}/api/v1/chat`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${API_KEY}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      messages: [{ role: "user", content: question }],
      mode: "routeplex-ai",
    }),
  });

  if (!response.ok) {
    console.error(`Error ${response.status}: ${await response.text()}`);
    return null;
  }

  const data = await response.json();
  console.log(
    `  Model: ${data.data.model_used} | Cost: $${data.data.usage.cost_usd}`
  );
  return data.data.output;
}

async function main() {
  const queries = [
    "What are the trending topics on tech news today?",
    "What is the current price of Bitcoin?",
  ];

  for (const q of queries) {
    console.log(`\nQ: ${q}`);
    const answer = await searchAndAnswer(q);
    console.log(`A: ${answer}`);
  }
}

main();
